lib/facility.rb

```diff
@@ -1,14 +1,72 @@
+require './lib/vehicle'
+require './lib/registrant'
+require './lib/facility_factory'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :collected_fees,
+              :registered_vehicles,
+              :daily_hours,
+              :holiday_closed
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(input)
+    @name = input[:name]
+    @address = input[:address]
+    @phone = input[:phone]
     @services = []
+    @collected_fees = 0
+    @plate_type = {}
+    @registered_vehicles = []
+    @daily_hours = input[:daily_hours]
+    @holiday_closed = input[:holiday_closed]
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def collect_fees(vehicle)
+    if vehicle.plate_type == :ev
+      @collected_fees  += 200
+    elsif vehicle.plate_type == :antique
+      @collected_fees += 25
+    else
+      @collected_fees += 100
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      @registered_vehicles << vehicle
+    else
+      return
+    end
+  end
+
+  def administer_written_test(registrant)
+    return false if !@services.include?('Written Test')
+    return false if !registrant.permit? || registrant.age < 16
```

Comment:
You can combine lines 50 and 51 actually. Just check all the conditions on one line and return false if they're not all satisfied.